src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -666,6 +668,51 @@ private Identifier ParseIdentifier(Token at = null)
             return new Identifier(at, tok);
         }
 
+        private TexlNode ParseStringInterpolation()
+        {
+            Contracts.Assert(_curs.TidCur == TokKind.StrInterpStart);
+
+            _curs.TokMove();
+
+            var args = new Stack<TexlNode>();
+            var trivia = new Stack<ITexlSource>();
+            var ops = new Stack<Token>();
+
+            while (_curs.TidCur != TokKind.StrInterpEnd)
+            {
+                switch (_curs.TidCur)
+                {
+                    case TokKind.StrLit:
+                        args.Push(new StrLitNode(ref _idNext, _curs.TokMove().As<StrLitToken>()));
+                        break;
+                    case TokKind.IslandStart:
+                    case TokKind.IslandEnd:
+                        var token = _curs.TokMove();
+                        trivia.Push(new TokenSource(token));
+                        ops.Push(new KeyToken(TokKind.Ampersand, token.Span));
+                        break;
+                    default:
+                        args.Push(ParseExpr(Precedence.None));
+                        break;
+                }
+            }
+
+            TexlNode binaryNode = null;
+            while (args.Count > 1)
```

Comment:
Should we be generating this as a bunch of binary ops, instead of a single callnode to Concatenate with the args as an ordered list? Either way works, just wondering why this choice?

Reply:
Honestly I don't know how to deal with all the parameters to the Concatenate function, for example the sources list and trivia list inputs, because we are generating a Call Node without one appearing in the source.

Reply:
Flat here could actually be important.  The nature of $".." usage could lead to abnormally long concatentations; and our binary op is inefficient. 

I think this would just be building up the arg list to Call().

Reply:
Rewrote the parser changes to use Concatenate instead of a chain of & operators. One consequence of this is that I had to comment out an assert statement in the constructor of IdentToken.cs because the length of the span is incorrect when generating that token with the text "Concatenate".

We have 3 options:

1. Leave it the way it is in this PR
2. Make a new function named $" and map that function name to Concatenate at a later stage
3. Implement a new AST node to fully represent this feature, not as syntactic sugar but as an actual language construct

Reply:
I'm leaning towards 1 at the moment. For 2, I don't think we have a place to do that cleanly, and for 3, we'd basically be replicating a specialized version of all the various CallNode visitors later on. Removing that check does seem a little odd, but I think the benefits of keeping this feature limited to Lexer/Parser outweigh the weirdness of an IdentToken that doesn't map to a literal ident in the source.

Reply:
Agreed

Reply:
Agree with 1. Definitely not 2. We can always switch to 3 later - thay may be useful for some IDE features.